plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2251,6 +2249,7 @@ bool producer_plugin_impl::block_is_exhausted() const {
 // -> Idle
 // --> Start block B (block time y.000) at time x.500
 void producer_plugin_impl::schedule_production_loop() {
+   _received_block = false;
```

Comment:
Is it possible a race condition that a block is received after `start_block_interrupted` is called but before `schedule_production_loop`?

Reply:
Yes, :(. Not really anything we can do about that.